persistence/postgres-persistence/src/main/java/com/netflix/conductor/postgres/dao/PostgresMetadataDAO.java

```diff
@@ -438,16 +440,21 @@ private void insertOrUpdateWorkflowDef(Connection tx, WorkflowDef def) {
         updateLatestVersion(tx, def.getName(), maxVersion);
     }
 
-    /**
-     * Query persistence for all defined {@link TaskDef} data.
-     *
-     * @param tx The {@link Connection} to use for queries.
-     * @return A new {@code List<TaskDef>} with all the {@code TaskDef} data that was retrieved.
-     */
     private List<TaskDef> findAllTaskDefs(Connection tx) {
-        final String READ_ALL_TASKDEF_QUERY = "SELECT json_data FROM meta_task_def";
-
-        return query(tx, READ_ALL_TASKDEF_QUERY, q -> q.executeAndFetch(TaskDef.class));
+        final String READ_ALL_TASKDEF_QUERY =
+                "SELECT modified_on, json_data FROM meta_task_def";
+        List<TaskDef> taskDefs = new ArrayList<>();
+        try (PreparedStatement statement = tx.prepareStatement(READ_ALL_TASKDEF_QUERY)) {
+            ResultSet rs = statement.executeQuery();
+            while (rs.next()) {
+                var taskDef = objectMapper.readValue(rs.getString("json_data"), TaskDef.class);
+                taskDef.setUpdateTime(rs.getTimestamp("modified_on").getTime());
+                taskDefs.add(taskDef);
+            }
+        } catch (Exception e) {
```

Comment:
Catching all Exceptions is not a good practice.
And also, logging on error level without re-throwing an exception is a bit weird.

How are exceptions handled in "executeAndFetch" ? like this ? 
I would: catch a more specific exception, log warn or error and then rethrow as unchecked exception

... the way it is now, if an error occurs, the user will receive an empty list (like there are no tasks) and would have to look into the log to find the error.

Reply:
done